backend/special_effects/conditional_effect.py

```diff
@@ -94,8 +94,7 @@ def update(self, state, active=False):
             active (bool): Whether or not the update is due to an action being
             performed.
         """
-        if active: return
-        if self.completed: return
+        if active or self.completed: return
         for condition, value in self.condition.items():
             if state.predicates[condition] != value:
                 state.special_effects.remove(self)
```

Comment:
continuing [this](https://github.com/portal-cornell/robotouille/pull/22#discussion_r1518506012) thread here - this here is what confused me. the Delayed and Repetitive special effects don't remove themselves so it is clearer how they work. the `self.completed` check in `update_special_effect` is to avoid adding the effect again while the one inside the special effect is to avoid passive updates. I feel like a `self.completed` check within the special effect should be enough. have you considered updating the special effect's `__eq__` function so that you check everything but `self.completed` for equality? I imagine each special effect is unique to its own object and there would never be a special effect that exists where the only difference is `self.completed`.

Reply:
Changed eq function to only check self.completed within special effect class

Reply:
Is it necessary to remove the special effect if any of the predicates aren't the correct value? Why not just return like before so this conditional effect remains until completed and isn't updated again?

Reply:
replace removal with return